### ISSUE TITLE
Rubocop --auto-correct generates invalid syntax

### ISSUE THREAD

**Issue by samphilipd** [bbatsov/rubocop#4364]
There is a line in one of my projects that `rubocop --auto-correct` mangles into nonsense.

## Steps to reproduce the problem

Here's the code:

```ruby
if tile and tile[:visible] and !tile[:mine] and (tile[:neighbourhood_count] == 0)
  puts "yup"
end
```

`rubocop --auto-correct` outputs the following:

```ruby
if tile && tile[:visible)](&& !tile[:mine)](&& (tile[:neighbourhood_count] == 0)
  puts "yup"
end
```

Which is not valid Ruby syntax.

## RuboCop version

```
$ rubocop -V
0.48.1 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-darwin16)
```

**Comment by rrosenblum** [comment]
I have been doing a bit of digging into this a bit. The issue is isolated to the `Style/AndOr` cop. The issue stems from the `correct_send` method. It looks like that method doesn't handle enumerable accessors properly. 

```ruby
if tile and tile[:visible]
  puts 'yup'
end
```
is corrected to 
```ruby
if tile && tile[:visible)](  puts 'yup'
end
```

```ruby
if tile and tile(:visible)
  puts 'yup'
end
```
is corrected to 
```ruby
if tile && tile(:visible)
  puts 'yup'
end
```

`correctable_send?` returns `true` for `tile[:visible]` and `false` for `tile(:visible)`. I might have a fix for this soon.

**Comment by Drenmi** [comment]
Looks like this is now correcting properly on current `master`. 🙂
